pkg/log/logger.go

```diff
@@ -86,3 +88,39 @@ func newZapLoggerTo(destWriter io.Writer, level LogLevel, opts ...zap.Option) *z
 	return zap.New(zapcore.NewCore(&kube_log_zap.KubeAwareEncoder{Encoder: enc, Verbose: level == DebugLevel}, sink, lvl)).
 		WithOptions(opts...)
 }
+
+type loggerCtx struct{}
+
+// CtxWithFields will add to provided context fields (key/value pairs)
+// which then will be logged by logger decorated with DecorateWithCtx function.
+// After adding logging fields it returns back enriched context.
+func CtxWithFields(ctx context.Context, keysAndValues ...interface{}) context.Context {
```

Comment:
why not just use `.Values`? Why include values in this function?

Reply:
We've followed up [here](https://github.com/kumahq/kuma/pull/7285)